### PR TITLE
✨ Add bottlerocket k8s settings support

### PR DESCRIPTION
**Pull request by abhinavmpandey08**
**What this PR does / why we need it**:
Adds support for configuring the following Kubernetes settings on Bottlerocket:
```yaml
   bottlerocket:
      kubernetes:
        maxPods: 50
        clusterDNSIPs:
          - 1.2.3.4
          - 4.3.2.1
        allowedUnsafeSysctls:
          - "net.core.somaxconn"
          - "net.ipv4.ip_local_port_range"
```

Also updated the unit tests for bottlerocket userdata generation to cover different test cases separately

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

Fixes https://github.com/aws/eks-anywhere/issues/5070

### CODE REVIEW

File: bootstrap/kubeadm/api/v1beta1/kubeadm_types.go [hunk 1]
```diff
@@ -185,6 +185,31 @@ type ClusterConfiguration struct {
 	// This is only for bottlerocket.
 	// +optional
 	BottlerocketCustomBootstrapContainers []BottlerocketBootstrapContainer `json:"bottlerocketCustomBootstrapContainers,omitempty"`
+
+	// BottlerocketSettings holds configuration for certain bottlerocket settings.
+	// This is only for bottlerocket.
+	// +optional
+	BottlerocketSettings *BottlerocketSettings `json:"bottlerocketSettings,omitempty"`
```
Comment (g-gaston) [inline bootstrap/kubeadm/api/v1beta1/kubeadm_types.go]:
```suggestion
	Bottlerocket *BottlerocketSettings `json:"bottlerocket,omitempty"`
```

File: bootstrap/kubeadm/api/v1beta1/kubeadm_types.go [hunk 2]
```diff
@@ -185,6 +185,31 @@ type ClusterConfiguration struct {
...
+// BottlerocketSettings define bottlerocket settings that can be configured on bottlerocket nodes.
+// This setting is ONLY for bottlerocket nodes.
+type BottlerocketSettings struct {
+	// Kubernetes holds the kubernetes settings for bottlerocket nodes.
+	Kubernetes *KubernetesSettings `json:"kubernetes,omitempty"`
+}
+
+// KubernetesSettings holds the settings for kubernetes on bottlerocket nodes.
+// This setting is ONLY for bottlerocket nodes.
+type KubernetesSettings struct {
```
Comment (g-gaston) [inline bootstrap/kubeadm/api/v1beta1/kubeadm_types.go]:
```suggestion
type BottlerocketKubernetesSettings struct {
```

File: bootstrap/kubeadm/api/v1beta1/kubeadm_types.go [hunk 3]
```diff
@@ -185,6 +185,31 @@ type ClusterConfiguration struct {
...
+	// MaxPods defines the maximum number of pods that can run on a node.
+	MaxPods int `json:"maxPods,omitempty"`
+
+	// AllowedUnsafeSysctls defines the list of unsafe sysctls that can be set on a node.
+	AllowedUnsafeSysctls []string `json:"allowedUnsafeSysctls,omitempty"`
+
+	// ClusterDNSIP defines the IP address of the DNS servers.
+	ClusterDNSIP []string `json:"clusterDNSIP,omitempty"`
```
Comment (g-gaston) [inline bootstrap/kubeadm/api/v1beta1/kubeadm_types.go]:
```suggestion
	// ClusterDNSIPs defines the IP addresses of the DNS servers.
	ClusterDNSIPs []string `json:"clusterDNSIPs,omitempty"`
```

**Review by g-gaston — COMMENTED** [review]
Just naming, rest looks good

**Comment by abhinavmpandey08** [comment]
> Just naming, rest looks good

resolved

**Review by g-gaston — APPROVED** [review]

**Pull request merged**